src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordHelpers.cs

```diff
@@ -407,10 +409,16 @@ public static bool IsDefaultEqualsOperator(IMethodBodyOperation operation)
             }
             else
             {
-                return SyntaxFactory.LiteralExpression(
-                    property.Type.NullableAnnotation == NullableAnnotation.Annotated
-                        ? SyntaxKind.NullLiteralExpression
-                        : SyntaxKind.DefaultLiteralExpression);
+                if (property.Type.NullableAnnotation == NullableAnnotation.Annotated)
+                {
+                    return SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression);
+                }
+                else
+                {
+                    // use type syntax in case of signature conflict
+                    return SyntaxFactory.DefaultExpression(property.Type.GenerateTypeSyntax(allowVar: false))
+                        .WithAdditionalAnnotations(Simplifier.Annotation);
+                }
```

Comment:
i feel like a helper for this exists somewhere.  can you look at the ParameterGenerator code (which likely defers to the ExpressionGenerator helper) to see if it has something that can be reused?

Reply:
@CyrusNajmabadi Is this what you're referring to?

https://github.com/dotnet/roslyn/blob/217eb9882e5bee0df02e05e8e3dc9b763b8c3da9/src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs#L3200-L3233